exporter/opentelemetry-exporter-otlp/src/opentelemetry/exporter/otlp/exporter.py

```diff
@@ -123,23 +131,42 @@ class OTLPExporterMixin(
         endpoint: OpenTelemetry Collector receiver endpoint
         credentials: ChannelCredentials object for server authentication
         metadata: Metadata to send when exporting
+        compression: Compression algorithm to be used in channel
     """
 
     def __init__(
         self,
         endpoint: str = "localhost:55680",
         credentials: ChannelCredentials = None,
         metadata: Optional[Tuple[Any]] = None,
+        compression: OTLPCompression = OTLPCompression.NO_COMPRESSION,
     ):
         super().__init__()
 
         self._metadata = metadata
         self._collector_span_kwargs = None
 
+        if compression == OTLPCompression.GZIP:
+            compression_algorithm = Compression.Gzip
+        else:
+            compression_str = os.environ.get(
+                "OTEL_EXPORTER_OTLP_COMPRESSION", None
+            )
+            if compression_str == OTLPCompression.GZIP:
```

Comment:
I don't think you can compare enum value with string like this, this will always be false. I'd recommend doing `OTLPCompression(compression_str)` to parse the string, handling the `ValueError` in the case it can't be parsed out (None or invalid)